src/meetbot-manager.ts

```diff
@@ -55,6 +73,8 @@ export async function spawnBot(url: string) {
 
 	// Create a new bot instance with the already created browser instance
 	const bot = new MeetBot(url, browser, allFeatures);
+	Bots.set(url, bot);
```

Comment:
why add a bot to the list of bots before it has joined the meet ? The bot gets added when the joined event is emitted.

Reply:
@20k-ultra this is an edge case I found in testing. 

From the time where we spawn a bot to the time the bot joins a meeting and the event is emitted - We are experiencing a state of limbo. 

In this period of time, the scheduler thinks that the bot hasn't spawned yet for the meet and therefore the scheduler attempts to start another bot. In a couple of second that it takes for the "join" event to be emitted almost 15+ bots are added to the queue for the same meet. 

Hence, my solution of add the bot to the Bots map ASAP

Reply:
That sounds like you're breaking the joined functionality. You're saying a bot has joined when it hasn't to prevent another process from doing something wrong. I strongly feel that we should not say a bot has joined when it is not in a meeting.

Reply:
I realized after that this is just adding a bot to the list of bots not emitting a joined event my bad. So this doesn't seem as critical, in fact, tracking bots that are "joining" or "dead" etc is useful so you're good.